src/api/app/controllers/webui/webui_controller.rb

```diff
@@ -27,7 +27,7 @@ def handle_unverified_request
   add_flash_types :success, :error
 
   rescue_from Pundit::NotAuthorizedError do |exception|
-    pundit_action = case exception.query.to_s
+    pundit_action = case exception.try(:query).to_s
```

Comment:
Anyway, I don't like throwing an exception in https://github.com/openSUSE/open-build-service/blob/master/src/api/app/policies/application_policy.rb#L6 because I would like to use Pundit view helper more often in the future (``policy(comment).destroy?``). However, with the exception it is not possible to use it when User is nobody. We would need to do always ``!User.current.is_nobody? && policy(comment).destroy? `` in this order to not call Pundit and raise the exception. Other solution (which I also don't really like) is to check for the user record in the policies, however, this would mean duplicate code and we would get a different flash message then (not stating that you're not logged in, just that you're not authorized).

@hennevogel ^^